vpp/clientconfigsrv.go

```diff
@@ -0,0 +1,155 @@
+	AppleID            string   `json:"appleId,omitempty"`
+	OrganizationIDHash string   `json:"organizationIdHash"`
+	Status             int      `json:"status"`
+	OrganizationID     int      `json:"organizationId"`
+	UID                string   `json:"uId"`
+	CountryCode        string   `json:"countryCode"`
+	Location           Location `json:"location"`
+	APNToken           string   `json:"apnToken"`
+	Email              string   `json:"email"`
+}
+
+// These specify options for the ClientConfigSrv
+type GetClientConfigSrvOptions func(*getClientConfigSrvOpts) error
+
+type getClientConfigSrvOpts struct {
+	SToken        string `json:"sToken"`
+	Verbose       bool   `json:"verbose,omitempty"`
+	ClientContext string `json:"clientContext,omitempty"`
+}
+
+// Verbose is an optional argument that can be added to GetClientConfigSrv
+func VerboseOption(verbose bool) GetClientConfigSrvOptions {
+	return func(opts *getClientConfigSrvOpts) error {
+		opts.Verbose = verbose
+		return nil
+	}
+}
+
+// ClientContext is an optional argument that can be added to GetClientConfigSrv
+func ClientContextOption(context string) GetClientConfigSrvOptions {
+	return func(opts *getClientConfigSrvOpts) error {
+		opts.ClientContext = context
+		return nil
+	}
+}
+
+// Gets ClientConfigSrv information
+func (c *Client) GetClientConfigSrv(opts ...GetClientConfigSrvOptions) (*ClientConfigSrv, error) {
+	// Set required and optional arguments
+	request := &getClientConfigSrvOpts{SToken: c.SToken}
+	for _, option := range opts {
+		if err := option(request); err != nil {
+			return nil, err
+		}
+	}
+
+	// Get the ClientConfigSrvURL
+	clientConfigSrvURL := c.VPPServiceConfigSrv.ClientConfigSrvURL
+
+	// Create the ClientConfigSrvURL request
+	req, err := c.newRequest("POST", clientConfigSrvURL, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "create ClientConfigSrv request")
+	}
+
+	// Make the request
+	var response ClientConfigSrv
+	err = c.do(req, &response)
+
+	return &response, errors.Wrap(err, "make ClientConfigSrv request")
+}
+
+// Gets the appleID field along with the standard information
+func (c *Client) GetClientConfigSrvVerbose() (*ClientConfigSrv, error) {
+	options := VerboseOption(true)
+	response, err := c.GetClientConfigSrv(options)
+	if err != nil {
+		return nil, errors.Wrap(err, "using verbose option")
+	}
+
+	return response, nil
+}
+
+// Gets the values that determine which mdm server is associated with a VPP account token
+func (c *Client) GetClientContext() (*ClientContext, error) {
+	// Get the ClientConfigSrv info
+	clientConfigSrv, err := c.GetClientConfigSrv()
+	if err != nil {
+		return nil, errors.Wrap(err, "get ClientContext request")
+	}
+
+	// Get the ClientContext string
+	var context = clientConfigSrv.ClientContext
+
+	// Convert the string to a ClientContext type
+	var clientContext ClientContext
+	decodeError := json.NewDecoder(strings.NewReader(context)).Decode(&clientContext)
```

Comment:
just keep `err = `, it's very rare to ever need a new error variable.

Reply:
I posted a commit for this, just this morning.